src/UI/Actions.hs

```diff
@@ -1052,7 +1054,10 @@ openCommand' s cmd =
               (tmpfileResource (view mhKeepTemp cmd))
               (\_ fp -> toProcessConfigWithTempfile (view mhMakeProcess cmd) fp) <$>
             entityToBytes e
-   in either (pure . flip setError s) (`runEntityCommand` s) mkConfig
+   in either
```

Comment:
Is it possible to simplify this? I couldn't find a solution, since `mkConfig` produces an `Either` while the `runExceptT` gives an `Either` wrapped in an `IO`.

Reply:
You might be able to simplify by having something like:
```haskell
mkConfig :: (MonadError Error m) => m (EntityCommand a FilePath)
```
And `throwError` instead of `Left`.  Then

```haskell
either handleError id $ runExceptT (mkConfig >>= runEntityCommand ec s)
```

(Something like that; just a sketch).

Reply:
Cheers! Will try it out.